hyperspy/io.py

```diff
@@ -178,24 +178,42 @@ def load(filenames=None,
         if len(filenames) > 1:
             messages.information('Loading individual files')
         if stack is True:
-            signal = []
             for i, filename in enumerate(filenames):
                 obj = load_single_file(filename,
                                        **kwds)
-                signal.append(obj)
-            signal = hyperspy.utils.stack(signal,
-                                          axis=stack_axis,
-                                          new_axis_name=new_axis_name,
-                                          mmap=mmap, mmap_dir=mmap_dir)
-            signal.metadata.General.title = \
-                os.path.split(
+                if i == 0:
+                    if isinstance(obj, (list, tuple)):
+                        n = len(obj)
+                    else:
+                        n = 1
+                    signals = [[] for j in xrange(n)]
+                else:
+                    if isinstance(obj, (list, tuple)):
```

Comment:
This `ValueError` will be raised if the different files do not contain the same number of signals. `n` is set from the first file loaded, and then is checked for each subsequent file. To make it less confusing, I think to add an exception message along the lines of "Files do not have the same number of sub-signals, so they cannot be stacked", and then throw in offending file names and numbers for details.